builtin/plugins/coin/coin.go

```diff
@@ -213,6 +257,97 @@ func mint(ctx contract.Context, to loom.Address, amount *loom.BigUInt) error {
 	return ctx.Set(economyKey, econ)
 }
 
+//Mint : to be called by CoinPolicyManager Responsible to mint coins as per various parameter defined
+func Mint(ctx contract.Context) error {
+	div := loom.NewBigUIntFromInt(10)
+	div.Exp(div, loom.NewBigUIntFromInt(18), nil)
+	var policy Policy
+	var amount *common.BigUInt
+	err := ctx.Get(policyKey, &policy)
+	if err != nil {
+		return errUtil.Wrap(err, "Failed to Get PolicyInfo")
+	}
+	changeRatioNumerator := loom.NewBigUIntFromInt(int64(policy.ChangeRatioNumerator))
+	changeRatioDenominator := loom.NewBigUIntFromInt(int64(policy.ChangeRatioDenominator))
+	blockHeight := loom.NewBigUIntFromInt(ctx.Block().Height)
+	baseAmount := loom.NewBigUIntFromInt(int64(policy.TotalSupply))
+	baseAmount.Mul(baseAmount, div)
+	blocksGeneratedPerYear := loom.NewBigUIntFromInt(int64(policy.BlocksGeneratedPerYear))
+	basePercentage := loom.NewBigUIntFromInt(int64(policy.BasePercentage))
+	operator := policy.Operator
+	_, modulus := big.NewInt(ctx.Block().Height).DivMod(big.NewInt(ctx.Block().Height), big.NewInt(int64(policy.BlocksGeneratedPerYear)), big.NewInt(int64(policy.BlocksGeneratedPerYear)))
+	year := blockHeight.Div(blockHeight, blocksGeneratedPerYear)
+	year = year.Add(year, loom.NewBigUIntFromInt(1))
+	var checkMintAmount = &types.BigUInt{
+		Value: *loom.NewBigUIntFromInt(0),
+	}
+	//Checks if minting amount is computed before or it is being computed for first time
+	err = ctx.Get(mintingAmountKey, checkMintAmount)
+	if err != nil && err != contract.ErrNotFound {
+		return err
+	}
+	//Minting Amount per block is set in ctx after minting amount is determined for year
+	if year.Cmp(loom.NewBigUIntFromInt(1)) == 0 {
+		//Determines minting amount at the beginning block of year or at block height at which minting is enabled
+		if modulus.Cmp(big.NewInt(1)) == 0 || err == contract.ErrNotFound {
+			//Minting Amount Computation for starting year
+			amount = baseAmount.Div(baseAmount, blocksGeneratedPerYear)
+			err = ctx.Set(mintingAmountKey, &types.BigUInt{
+				Value: *amount,
+			})
+			if err != nil {
+				return err
+			}
+		}
+	} else {
+		if modulus.Cmp(big.NewInt(1)) == 0 || err == contract.ErrNotFound {
+			//Operator Support to support different inflation ratio patterns for different year
+			if strings.EqualFold("div", operator) {
+				changeRatioDenominator = changeRatioDenominator.Mul(changeRatioDenominator, year)
+				basePercentage = basePercentage.Mul(basePercentage, changeRatioNumerator)
+			} else if strings.EqualFold("exp", operator) {
+				changeRatioNumerator = changeRatioNumerator.Exp(changeRatioNumerator, year, nil)
```

Comment:
In case of `exp` we dont need to change numerator only denominator

Reply:
Also, change naming convention. basePercentage is what we read from config, and then depending upon which year it is, we update it

Reply:
Check if we try to `div` by zero.

Reply:
Suppose base percentage is - 10
Change ratio of base percentage is - 2/3 
For consecutive years --> Base percentage will be as follows - 
1) Base percentage - 10 * 2/3 - ~6.6
2) Base percentage - 10 * 4/9 - ~4.4
So, you need to take exponential of both numerator and denominator

Reply:
I am doing input validation of denominator that it is not equal to zero, so practically denominator will never be zero.
But will add a check, here as well.